run-make-check.sh

```diff
@@ -12,9 +12,24 @@
 #  version 2.1 of the License, or (at your option) any later version.
 #
 
-#
-# Return MAX(1, (number of processors / 2)) by default or NPROC
-#
+trap clean_up_after_myself EXIT
```

Comment:
Trapping ERR would be nice too

Reply:
i don't think we should trap ERR, as quite a lot of commands could fail here, for example `test -f $ORIGINAL_CCACHE_CONF` might fail.

Reply:
As long as we do tweak the user env and restore it after a build, we shall be able to restore it also when something fails to get a consistent user env.